### PR TITLE
Feat/components

### PR DESCRIPTION
**Pull request by TranLeKhanhDuyen**

### CODE REVIEW

File: mega-market/src/components/SearchBar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import '@components/SearchBar/SearchBar.css'
+
+import searchIcon from '@assets/images/icons/search.svg'
+import listIcon from '@assets/images/icons/list.svg'
```
[the remark this answers is not in the document]
Comment (tuanthaiquang-agilityio) [reply mega-market/src/components/SearchBar/index.tsx:4]: Create index.ts in the folder component to export together

File: mega-market/src/components/SearchBar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import TextField from '@components/TextField'
+
+interface ISearchBarProps {
+  icon?: string
+  placeholder?: string
+  iconLeft?: string
+  iconRight?: string
+  onChange?: (value: string) => void
+  onClose?: () => void
```
Comment (tuanthaiquang-agilityio) [inline mega-market/src/components/SearchBar/index.tsx:13]: Why all props are optional?

File: mega-market/src/components/Select/Select.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  { value: 'groceriesB', label: 'Groceries B' },
+  { value: 'groceriesC', label: 'Groceries C' },
+  { value: 'groceriesD', label: 'Groceries D' }
+]
+
+export const Default: Story = {
+  args: {
+    options: options,
+    onSelect: (value: string) => {
+      console.log('Groceries:', value)
```
Comment (tuanthaiquang-agilityio) [inline mega-market/src/components/Select/Select.stories.tsx:23]: console.log?
Please re-read storybook

File: mega-market/src/components/Select/index.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    setIsOpen(false)
+  }
+
+  const selectButtonProps: IIconTextButtonProps = {
+    size: 'sm',
+    icon: selectIcon || '',
+    title: selectedOption ? selectedOption.label : 'Groceries',
+    iconPosition: 'right',
+    additionalClass: 'select',
+    onClick: () => setIsOpen(!isOpen)
```
Comment (tuanthaiquang-agilityio) [inline mega-market/src/components/Select/index.tsx:34]: Create handleToggleSelect

File: mega-market/src/components/Select/index.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    <div className='select-container'>
+      <IconTextButton {...selectButtonProps} />
+      {isOpen && (
+        <div className='options-container'>
+          {options.map((option) => (
+            <div
+              key={option.value}
+              className='option'
+              onClick={() => handleSelectOption(option)}
+            >
```
Comment (tuanthaiquang-agilityio) [inline mega-market/src/components/Select/index.tsx:47]: div can wrap a string?

File: mega-market/src/components/TextField/index.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  const error = validate ? validate(value || '') : undefined
+
+  return (
+    <>
+      <label className={`label-input ${!isShowLabel && 'hide'}`}>{label}</label>
+
+      <div className='input-container'>
+        {iconLeft && <img src={iconLeft} alt='icon' />}
```
Comment (tuanthaiquang-agilityio) [inline mega-market/src/components/TextField/index.tsx:39]: Double check alt and can we use iconLeft is a ReactNode?

File: mega-market/src/components/TextField/index.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        <input
+          type='text'
+          className={`text-field ${additionalClass}`}
+          value={value}
+          placeholder={placeholder}
+          onChange={handleChangeInput}
+        />
+        {iconRight && <img src={iconRight} alt='icon' />}
```
Comment (tuanthaiquang-agilityio) [inline mega-market/src/components/TextField/index.tsx:47]: Same here

File: mega-market/src/stories/Button.tsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  /**
+   * Optional click handler
+   */
+  onClick?: () => void;
+}
+
+/**
+ * Primary UI component for user interaction
+ */
+export const Button = ({
```
Comment (tuanthaiquang-agilityio) [inline mega-market/src/stories/Button.tsx]: Why do we have a button in stories?

File: mega-market/src/stories/Configure.mdx [hunk 9]
```diff
@@ -0,0 +1,364 @@
+import { Meta } from "@storybook/blocks";
```
Comment (tuanthaiquang-agilityio) [inline mega-market/src/stories/Configure.mdx]: mdx file?

File: mega-market/src/stories/Header.tsx [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+interface HeaderProps {
+  user?: User;
+  onLogin?: () => void;
+  onLogout?: () => void;
+  onCreateAccount?: () => void;
+}
+
+export const Header = ({ user, onLogin, onLogout, onCreateAccount }: HeaderProps) => (
```
Comment (tuanthaiquang-agilityio) [inline mega-market/src/stories/Header.tsx]: Why do we have components in here?

File: mega-market/src/stories/page.css [hunk 11]
```diff
@@ -0,0 +1,69 @@
+.storybook-page {
```
Comment (tuanthaiquang-agilityio) [inline mega-market/src/stories/page.css]: Why we need this? Can we remove all css stories?

File: mega-market/src/styles/utils/variables.css [hunk 12]
```diff
@@ -21,4 +21,5 @@
   --text-tertiary-color: #249b3e;
 
   --white-color: #ffffff;
```
Comment (tuanthaiquang-agilityio) [inline mega-market/src/styles/utils/variables.css:24]: #fff

File: mega-market/src/styles/utils/variables.css [hunk 13]
```diff
@@ -21,4 +21,5 @@
...
+  --black-color: #000000;
```
Comment (tuanthaiquang-agilityio) [inline mega-market/src/styles/utils/variables.css:25]: #000

File: mega-market/src/components/HeadLine/index.tsx [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+export interface IHeadline extends React.HTMLAttributes<HTMLDivElement> {
+  title: string
+  subTitle?: string
+  additionalClass?: string
+}
+
+const HeadLine = ({ title, subTitle, additionalClass }: IHeadline) => {
+  return (
+    <div className='headline-container'>
+      <span className={`headline headline-title ${additionalClass} `}>
```
[the remark this answers is not in the document]
Comment (TranLeKhanhDuyen) [reply mega-market/src/components/HeadLine/index.tsx:12]: Here I use the span tag because I want these two elements to be on the same line

**Pull request merged**